src/app/features/auth/auth/auth.component.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  inject,
+  OnInit,
+  OnDestroy,
+} from '@angular/core';
+import { NonNullableFormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+import { MatInputModule } from '@angular/material/input';
+import { MatIconModule } from '@angular/material/icon';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { NgIf } from '@angular/common';
+import { Subscription } from 'rxjs';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-auth',
+  standalone: true,
+  imports: [ReactiveFormsModule, MatInputModule, MatIconModule, MatCheckboxModule, NgIf],
+  templateUrl: './auth-component.html',
+  styleUrls: ['./auth.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AuthComponent implements OnInit, OnDestroy {
+  private builder = inject(NonNullableFormBuilder);
+  private router = inject(Router);
+
+  form = this.createForm();
+  emailNotFocused = false;
+  passwordNotFocused = false;
+  isFormValid = true;
```

Comment:
Dlaczego `isFormValid` jest ustawione domyślnie na true?

Reply:
po co w sumie potrzebujemy tego pola skoro mamy obiekt formularza gdzie mamy dostep do pola `this.form.valid` ?

Reply:
W sumie prawda. Nie pomyślałem o tym. Zmieniłem teraz, aby było wykorzystane this.form.valid i usunąłem zbędne metody oraz pole isFormValid.